src/backend/expungeservice/expunger/analyze.py

```diff
@@ -118,3 +118,182 @@ def __init__(self, name, dob, cases):
+        client: A Client instance
+    """
+
+    def __init__(self, client):
+        self.client = client
+
+    def _is_charge_level(charge, type_, class_):
+        check = 'Is the charge a {}'.format(type_)
+        if class_:
+            check += ' class {}'.format(class_)
+
+        result = (charge.level.type_ ==  type_ and
+                  (not class_ or charge.level.class_ == class_))
+        return Check(check=check, result=result)
+
+    def _is_charge_statute(charge, statute):
+        check = 'Does the charge fall under statute: ' + str(statute)
+        # TODO implement this
+        return Check(check=check, result=True)
+
+    def _is_charge_in_statute_list(charge, statutes, desc):
+        check = ('Does the charge fall under any of these statutes: ' +
+                 ','.join(str(statute) for statute in statutes))
+        # TODO implement this
+        return Check(check=check, result=False, check_desc=desc)
+
+    def _is_charge_sex_crime(charge):
+        # TODO update
+        _statutes_sex_crimes = []
+        return RecordAnalyzer._is_charge_in_statute_list(
+            charge, _statutes_sex_crimes, 'Is the charge a sex crime')
+
+    def _is_charge_traffic_crime(charge):
+        # TODO update
+        _statutes_traffic_crimes = []
+        return RecordAnalyzer._is_charge_in_statute_list(
+            charge, _statutes_traffic_crimes, 'Is the charge a traffic crime')
+
+    def _have_open_case(self):
+        check = 'Is there a open case for the client'
+        result = any([case.state == CaseState.OPEN for case in self.client.cases])
+        return Check(check=check, result=result)
+
+    """
+    Run Time Eligibility analysis on the client and their records
+
+    TODO: make it return analysis for each charge as well (which is supposed
+    to be an update on eligibility date and relevan statutes)
+
+    Returns:
+        An Result instance
+    """
+    def time_eligibility(self):
+        checks = []
+
+        checks.append(self._have_open_case())
+        if checks[-1].result:
+            return Result(ResultCode.OPEN_CASE, checks)
+
+        # TODO implement the rest
+
+        return Result(ResultCode.NO_ACTION)
+
+    """
+    Run Type Eligibility analysis on a charge
+
+    Args:
+        charge: A Charge instance
+
+    Returns:
+        A Result instance
+    """
+    def type_eligibility(self, charge):
+        checks = []
+
+        checks.append(RecordAnalyzer._is_charge_level(charge, 'Felony', 'A'))
+        if checks[-1].result:
+            return ResultInElig137(checks=checks)
+
+        checks.append(RecordAnalyzer._is_charge_sex_crime(charge))
+        if checks[-1].result:
+            return ResultInElig137(checks=checks)
+
+        checks.append(RecordAnalyzer._is_charge_traffic_crime(charge))
+        if checks[-1].result:
+            return ResultInElig137(checks=checks)
+
+        # TODO add remaining checks
+
+        return Result(ResultCode.FURTHER_ANALYSIS, checks)
```

Comment:
Let's discuss this method next time we meet so that I can get a better understanding of the approach being taken.

Reply:
Sure but it would be in about a month. This function basically runs through different checks and appends them to a list which would be returned as part of the type eligibility analysis.